File: growing-shop-java/src/main/java/com/example/growingshop/domain/product/domain/ProductCategory.java
package com.example.growingshop.domain.product.domain;

import com.example.growingshop.domain.category.domain.Category;
import com.example.growingshop.domain.company.domain.Company;
import com.example.growingshop.global.validator.StringChecker;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.util.ArrayList;
import java.util.List;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class ProductCategory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    private Product product;

    @ManyToOne
    private Category category;
}

File: growing-shop-domain/src/main/kotlin/com/example/domain/product/ProductCategory.kt
package com.example.domain.product

import com.example.domain.category.Category
import com.example.domain.common.IdentifiableEntity
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.ManyToOne

@Entity
class ProductCategory(
    @ManyToOne(fetch = FetchType.LAZY)
    val product: Product,

    @ManyToOne
    val category: Category
) : IdentifiableEntity()
